lib/internal/Magento/Framework/App/DeploymentConfig/Reader.php

```diff
@@ -117,7 +117,7 @@ public function load($fileKey = null)
                 }
             }
         }
-        return $result ?: [];
+        return is_array($result) ? $result : [];
```

Comment:
What do you think about throwing an exception instead of silently ignoring an invalid configuration files?

I mean, right after the configuration file inclusion.

Reply:
But that would still block the deployment process as described in #13090

Reply:
The current behavior when `$fileKey` is empty is actually also silently ignoring invalid files: https://github.com/barryvdh/magento2/blob/e4d857fe63717d24da7b3fbeaa44504fcd866632/lib/internal/Magento/Framework/App/DeploymentConfig/Reader.php#L115
So this would bring it inline with that behaviour.

Or do you want me to add check after both includes to throw an exception?

Reply:
It's fine to block deployment process in case of invalid data provided. Otherwise it would be pretty nontrivial to debug what went wrong.

Yeah, looks like `is_array` was randomly introduced in https://github.com/barryvdh/magento2/commit/02596b720fab5bd01fbc8026a1392c4971f78fed#diff-db4cf46912b0aad3f54619ff946b9ac8R115, please throw proper exception in both cases when file data is invalid.

Reply:
Better?